wallet/update.go

```diff
@@ -371,15 +371,25 @@ func (sw *SingleAddressWallet) UpdateChainState(tx UpdateTx, reverted []chain.Re
 			ID:     cru.Block.ID(),
 			Height: cru.State.Index.Height + 1,
 		}
-		if err := revertChainUpdate(tx, revertedIndex, sw.addr, cru); err != nil {
-			return err
+		sw.mu.Lock()
+		err := revertChainUpdate(tx, revertedIndex, sw.addr, cru)
+		if err != nil {
```

Comment:
At this point we could make `revertChainUpdate` and `applyChainState` methods `SingleAddressWallet` since they need to lock it and update its field.

Reply:
That’s true. Do you see a reason to keep UpdateChainState in that instance?

Reply:
I think it's fine to keep it since `UpdatesSince` also returns both apply and revert updates. Saves us the loops in `renterd` and `hostd`. But I'm not too attached.